src/main.cpp

```diff
@@ -2080,6 +2081,50 @@ void PartitionCheck(bool (*initialDownloadCheck)(), CCriticalSection& cs, const
     }
 }
 
+// Protected by cs_main
+static VersionBitsCache versionbitscache;
+
+int32_t ComputeBlockVersion(const CBlockIndex* pindexPrev, const Consensus::Params& params)
```

Comment:
Perhaps it would be useful to have a 'test' softfork that is disabled (end
time before genesis, for example) everywhere but regtest, and enables some
very easy to test condition (e.g. coinbase must have exactly one output)?